enaml/qt/qt_mpl_canvas.py

```diff
@@ -73,27 +76,48 @@ def _refresh_mpl_widget(self):
         """ Create the mpl widget and update the underlying control.
 
         """
-        # Delete the old widgets in the layout, it's just shenanigans
-        # to try to reuse the old widgets when the figure changes.
         widget = self.widget
         layout = widget.layout()
-        while layout.count():
-            layout_item = layout.takeAt(0)
-            layout_item.widget().deleteLater()
-
-        # Create the new figure and toolbar widgets. It seems that key
-        # events will not be processed without an mpl figure manager.
-        # However, a figure manager will create a new toplevel window,
-        # which is certainly not desired in this case. This appears to
-        # be a limitation of matplotlib. The canvas is manually set to
-        # visible, or QVBoxLayout will ignore it for size hinting.
+
+        # Create the new canvas and toolbar widgets if suitable ones have
+        # not been created.
+        # The canvas is manually set to visible, or QVBoxLayout will
+        # ignore it for size hinting.
         figure = self.declaration.figure
         if figure:
```

Comment:
What happens if the `figure` has been set to `None` in Enaml? Is the old stale figure still displayed? If so, that should be fixed.